xml/1.3/modules/ROOT/pages/xml-schema-validation.adoc

```diff
@@ -44,34 +69,77 @@ The `XML-MODULE:SCHEMA_NOT_HONOURED` error is a complex one. Because the validat
+. Set the *Collection* field to the expression that splits the payload into individual pieces, for example, `#[error.errorMessage.payload]`.
+. Drag a *Logger* component into the *For Each* scope component to log the XML module error.
+. Set the *Message* field to the expression that contains the `SchemaViolation` objects that describe the message, for example, `#['At line: $(payload.lineNumber), column: $(payload.columnNumber) -> $(payload.description)']`.
+. Save your changes.
+. Click the project name in *Package Explorer* and then click *Run > Run As > Mule Application*.
+. Navigate to the Console view to read the logger message:
++
+```
+ERROR 2018-02-16 14:35:45,722 [[MuleRuntime].cpuIntensive.01: [SchemaValidationTestCase#extractErrorsUsingExpressions].extractErrorsFromException.CPU_INTENSIVE @411e886b] org.mule.runtime.core.internal.processor.LoggerMessageProcessor: At line: -1, column: -1 -> cvc-complex-type.2.4.a: Invalid content was found starting with element 'fail'. One of '{used}' is expected.
+```
+.XML Module Manage validation error flow
+image::xml-validate-schema-3.png[The XML Module Manage validation error flow]
+
+In the XML editor, the `<on-error-propagate type="XML-MODULE:SCHEMA_NOT_HONOURED">` configuration looks like this:
 
 [source,xml,linenums]
 ----
  <flow name="extractErrorsFromException">
     <try>
-        <xml-module:validate-schema schemas="schema.xsd" /> // <1>
+        <xml-module:validate-schema schemas="schema.xsd" />
         <error-handler>
-            <on-error-propagate type="XML-MODULE:SCHEMA_NOT_HONOURED"> // <2>
+            <on-error-propagate type="XML-MODULE:SCHEMA_NOT_HONOURED">
                 <foreach collection="#[error.errorMessage.payload]">
-                    <logger level="ERROR" message="#['At line: $(payload.lineNumber), column: $(payload.columnNumber) -> $(payload.description)']" /> // <3>
+                    <logger level="ERROR" message="#['At line: $(payload.lineNumber), column: $(payload.columnNumber) -> $(payload.description)']" />
                 </foreach>
             </on-error-propagate>
         </error-handler>
     </try>
 </flow>
 ----
 
+
+== Troubleshoot Validation Against a Schema Issues
+
+When using the XML module to validate against a schema that has references to other local schema files, the validation can fail with the error:
+
 ```
-ERROR 2018-02-16 14:35:45,722 [[MuleRuntime].cpuIntensive.01: [SchemaValidationTestCase#extractErrorsUsingExpressions].extractErrorsFromException.CPU_INTENSIVE @411e886b] org.mule.runtime.core.internal.processor.LoggerMessageProcessor: At line: -1, column: -1 -> cvc-complex-type.2.4.a: Invalid content was found starting with element 'fail'. One of '{used}' is expected.
+The supplied schemas were not valid. schema_reference: Failed to read schema document NMVS_Composite_Types.xsd, because file access is not allowed due to restriction set by the accessExternalSchema property
```

Comment:
I can't tell which part of this, if any, is system output, and the second sentence looks like a mixture?
Best guess:
```suggestion
The supplied schemas are not valid, because file access is not allowed due to restrictions set by the `accessExternalSchema` property.
```

Reply:
Check how the sentence works with the following sentence.